File: Backend/SOS-Clean-House-back/src/main/java/com/SOS_Clean_House_back/model/Servico.java
package com.SOS_Clean_House_back.model;

import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Cascade;

import java.util.List;

@Entity
@AllArgsConstructor @NoArgsConstructor @Data @Builder


public class Servico {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer Id;
    private String valorDiario4H;
    private String valorDiario8H;
    private String tempoReserva;
    private String observacoes;
    private String servico;
    private List<String> diasDaSemana;
    private List<String> atividades;

    @ManyToOne()
    @JoinColumn(name = "prestador_id")
    private Prestador prestador;

    @OneToMany(mappedBy = "servico")
    private List<Horario> horarios;

}

File: Frontend/SOS-Clean-House-front/src/app/componentes/interface/Servico.ts
import { Agendamento } from "./Agendamento";
import { EnderecoServico } from "./EnderecoServico";
import { Horario } from "./Horario";
import { Prestador } from "./Prestador";
export interface Servico {
  id: number | undefined;
  valorDiario4H: string;
  valorDiario8H: string;
  tempoReserva: string;
  observacoes: string;
  servico: string;
  diasDaSemana: string[];
  atividades: string[];
  prestador: Prestador;
  horario: Horario;
  agendamento?: Agendamento;
  enderecoServico?: EnderecoServico;
  agendamentosRelacionados?: Agendamento[];
  }